components/script/dom/htmlimageelement.rs

```diff
@@ -426,45 +427,45 @@ fn image_dimension_setter(element: &Element, attr: Atom, value: u32) {
 // }
 
 fn parse_a_sizes_attribute(input: DOMString, width: Option<u32>)-> Result<Size,()>{
-    
-
-  //  let mut iter = Parser::new(input);
-
+   
+/*    let s_Size = Size{
+        length: Au,
+        expression:None
+    };*/
+    //extracting each media element from the string 
     let unparsed_sizes_list = input.deref().split(',').collect::<Vec<_>>();
-
+    //computing the last element calc() 
     let last_component = unparsed_sizes_list[unparsed_sizes_list.len()-1];
     let mut parser_last = Parser::new(last_component);
     let size_length = parser_last.try(Length::parse_non_negative);
-    match size_length {
+
+    //removing last element 
+    unparsed_sizes_list.remove(unparsed_sizes_list.len()-1);
+/*    match size_length {
         Ok(len) => {
-            let s_Size = Size{
+             s_Size{
                 length: len,
                 expression: None 
             };
         },
         Err(e) => {}
-    }
+    }*/
+
+    //checking other media elements to obtain expression
     for unparsed_size in unparsed_sizes_list{
         let whitespace = unparsed_size.chars().rev().take_while(|c| util::str::char_is_whitespace(*c)).count();
-        let trimmed = unparsed_size.chars().take(unparsed_size.chars().count() - whitespace);
+        let trimmed: String = unparsed_size.chars().take(unparsed_size.chars().count() - whitespace).collect();
         
+        let mut parse_each = Parser::new(&trimmed);
+        let each_media_query = parse_each.try(MediaQuery::parse);
```

Comment:
I'll point out that a media query is different than a media expression. We don't want to try to parse a whole media query here; I mentioned `Expression::parse` for a reason.